pkg/controller/account/ec2.go

```diff
@@ -58,14 +81,59 @@ func (r *ReconcileAccount) InitializeRegion(reqLogger logr.Logger, account *awsv
 		AwsRegion:               region,
 	})
 	if err != nil {
-		connErr := fmt.Sprintf("Unable to connect to region: %s when attempting to initialize it", region)
+		connErr := fmt.Sprintf("unable to connect to region %s when attempting to initialize it", region)
 		reqLogger.Error(err, connErr)
 		// Notify Error channel that this region has errored and to move on
 		ec2Errors <- connErr
 
 		return err
 	}
 
+	// If the quota is 0, there was an error and we cannot act on it
+	if vCPUQuota != 0 {
+		// Check if a request is necessary
+		// If there are errors, this will return false, and will not continue to try
+		// to set the quota
+		quotaIncreaseRequired, err = getVCPUQuota(awsClient, vCPUQuota)
+		if err != nil {
+			reqLogger.Error(err, "failed retriving current vCPU quota from AWS")
+		}
+	}
+
+	if quotaIncreaseRequired {
+		reqLogger.Info("vCPU quota increase required", "region", region)
+		caseID, err = checkQuotaRequestHistory(awsClient, vCPUQuota)
+		if err != nil {
+			reqLogger.Error(err, "failed retriving quota change history")
+		}
+
+		// If a Case ID was found, log it - the request was already submitted
+		if caseID != "" {
+			reqLogger.Info("found matching quota change request", "caseID", caseID)
+		}
+
+		// If there is not matching request already,
+		// and there were no errors trying to retrieve them,
+		// then request a quota increase
+		if caseID == "" && err == nil {
+			reqLogger.Info("submitting vCPU quota increase request", "region", region)
+			caseID, err = setVCPUQuota(awsClient, vCPUQuota)
+			if err != nil {
+				reqLogger.Error(err, "failed requesting vCPU quota increase")
+			}
+		}
+
+		// If the caseID is set, a quota increase was requested, either just now or previously. Log it.
+		// Can't update account conditions from within the asyncRegionInit goroutine, because
+		// the account is being updated elsewhere and will conflict.
+		if caseID != "" {
+			reqLogger.Info("quota increase request submitted successfully", "region", region, "caseID", caseID)
+		}
+	}
+
+	// Unset err to be explicit for use below
+	err = nil
```

Comment:
This is weird. Wouldn't L137 overwrite it no matter what?

Reply:
Just trying to be clear, but yes, you are right.  Removed.